### DISCUSSION TITLE
buttonMode roll out not working correctly on Linux x64

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#3481] [Q&A]
Hello,

I have a strange behavior on Ubuntu x64, using `buttonMode` the hand cursor is displayed when I roll out the blue square :
```
package
{
	import flash.display.Sprite;
	import flash.display.StageAlign;
	import flash.display.StageScaleMode;
	import flash.events.Event;
	import flash.text.TextField;
	
	public class Main extends Sprite
	{
		public function Main()
		{
			this.addEventListener(Event.ADDED_TO_STAGE, addedToStageHandler);
		}
		
		private function addedToStageHandler(event:Event):void
		{
			this.removeEventListener(Event.ADDED_TO_STAGE, addedToStageHandler);
			stage.scaleMode = StageScaleMode.NO_SCALE;
			stage.align = StageAlign.TOP_LEFT;
			
			var textField:TextField = new TextField();
			textField.width = 300;
			this.addChild(textField);
			
			var sprite:Sprite = new Sprite();
			sprite.graphics.beginFill(0x0000FF);
			sprite.graphics.drawRect( 0, 0, 100, 100);
			sprite.graphics.endFill();
			sprite.y = textField.y + textField.height + 20;
			sprite.buttonMode = true;
			this.addChild( sprite );
		}
	}
}
```
![PXL_20240919_193530298](https://github.com/user-attachments/assets/6bcf1e79-367b-43e9-8890-a909c49cad3f)

but I don't have this behavior on Ubuntu ARM64

![PXL_20240919_202956309](https://github.com/user-attachments/assets/9fe3c44f-85ee-49bc-98c0-3969fa0b7815)

Thanks

**Comment by itlancer** [comment]
For me it works fine with Ubuntu 22.04 x86_64.
But may be it related somehow to:
https://github.com/airsdk/Adobe-Runtime-Support/issues/1897
https://github.com/airsdk/Adobe-Runtime-Support/issues/193
https://github.com/airsdk/Adobe-Runtime-Support/issues/194
https://github.com/airsdk/Adobe-Runtime-Support/issues/1123
https://github.com/airsdk/Adobe-Runtime-Support/issues/3328

**Reply by pol2095** [reply]
@itlancer this bug occur on Ubuntu 24.04.1 LTS x86_64 and not using 22.04 x86_64, can you try with Ubuntu 24.04.1 ?

**Reply by itlancer** [reply]
@pol2095, @ajwfrost 
For my similar cases with complex AIR application:
With Ubuntu x86_64 22.04.4 LTS with X11 window manager - **works fine**.
With Ubuntu x86_64 22.04.4 LTS with Wayland window manager - **works fine**.
With Ubuntu x86_64 24.04.1 LTS with X11 window manager - **works fine**.

With Ubuntu x86_64 24.04.1 LTS with Wayland window manager - **strange behavior**.
Sometimes (I cannot figure out when and why) mouse cursor didn't update it state correctly. Sometimes it doens't show "finger" on roll over to `Sprite` with `buttonMode`. Sometimes it didn't change state on roll over to `TextField`. Sometimes it "stays" in previous state even after roll out from `TextField` or `Sprite` with `buttonMode`.
So yes, something definitely goes wrong.

**Reply by pol2095** [reply]
Harman in AIR SDK 51.2 migrate to GTK3 (GTK2 is not maintained), I don't know if this will fix the problem, GTK3 supports Wayland natively...

**Reply by pol2095** [reply]
I asked xorg xserver, it seems to be a bug in Gnome Mutter...

**Comment by pol2095** [comment]
If I update a TextField when mouse move, it work correctly :

```
package
{
	import flash.display.Sprite;
	import flash.display.StageAlign;
	import flash.display.StageScaleMode;
	import flash.events.Event;
	import flash.text.TextField;
	import flash.events.MouseEvent;
	
	public class Main extends Sprite
	{
		private var textField:TextField = new TextField();
		
		public function Main()
		{
			this.addEventListener(Event.ADDED_TO_STAGE, addedToStageHandler);
		}
		
		private function addedToStageHandler(e:Event):void
		{
			this.removeEventListener(Event.ADDED_TO_STAGE, addedToStageHandler);
			stage.scaleMode = StageScaleMode.NO_SCALE;
			stage.align = StageAlign.TOP_LEFT;
			
			textField.border = true;
			this.addChild(textField);
			
			var sprite:Sprite = new Sprite();
			sprite.graphics.beginFill(0x0000FF);
			sprite.graphics.drawRect( 0, 120, 100, 100);
			sprite.graphics.endFill();
			sprite.buttonMode = true;
			this.addChild( sprite );
			
			stage.addEventListener(MouseEvent.MOUSE_MOVE, mouseMoveHandler);
		}
		
		private function mouseMoveHandler(event : MouseEvent) : void
		{
			textField.text = event.stageX + ", "+ event.stageY;
		}
	}
}
```

**Comment by pol2095** [comment]
This bug doesn't occur when I launch Ubuntu with Xorg, only occur with Wayland, this bug occurs on Ubuntu 24.04.1 LTS x64 and not on Ubuntu 22.04.5 x64 (I tried a live usb using Wayland).

**Comment by pol2095** [comment]
The bug is fixed on Gnome 47.2 Wayland (Fedora 41).
